wasm/validation.cpp

```diff
@@ -682,6 +686,33 @@ tl::expected<void, ValidationError> validate(Module const &m) {
         }
     }
 
+    // https://webassembly.github.io/spec/core/valid/modules.html#start-function
+    if (m.start_section.has_value()) {
+        // Ensure function and type sections exist
+        if (!m.function_section.has_value()) {
+            return tl::unexpected{ValidationError::FunctionSectionUndefined};
+        }
+
+        if (!m.type_section.has_value()) {
+            return tl::unexpected{ValidationError::TypeSectionUndefined};
+        }
+
+        // Ensure start function has valid index into function and type section
+        if (m.start_section->start >= m.function_section->type_indices.size()) {
+            return tl::unexpected{ValidationError::StartFunctionInvalid};
+        }
+
+        if (m.start_section->start >= m.type_section->types.size()) {
+            return tl::unexpected{ValidationError::StartFunctionTypeInvalid};
```

Comment:
This isn't hit by the tests. Can you add a test w/ an empty type section? I /think/ that would hit this..